src/StackExchange.Utils.Http/DefaultHttpClientPool.cs

```diff
@@ -25,14 +26,20 @@ public class DefaultHttpClientPool : IHttpClientPool
         /// </summary>
         /// <param name="builder">The builder to get a request from.</param>
         /// <returns>The found or created <see cref="HttpClient"/> from the pool.</returns>
-        public HttpClient Get(IRequestBuilder builder) => ClientPool.GetOrAdd(new HttpClientCacheKey(builder.Timeout), CreateHttpClient);
+        public HttpClient Get(IRequestBuilder builder) => ClientPool.GetOrAdd(new HttpClientCacheKey(builder.Timeout, builder.Proxy), CreateHttpClient);
 
         private HttpClient CreateHttpClient(HttpClientCacheKey options)
         {
             var handler = new HttpClientHandler
             {
                 UseCookies = false
             };
+            if (options.Proxy != null)
+            {
+                handler.UseProxy = options.Proxy != null;
```

Comment:
`handler.UseProxy = true;`

Reply:
I think you want to keep `handler.UseProxy = false` by default, because otherwise it'll use the system proxy, which is a behavior change from what it is now: https://github.com/dotnet/corefx/blob/d3942d4671919edb0cca6ddc1840190f524a809d/src/System.Net.Http/src/System/Net/Http/HttpClientHandler.Windows.cs#L454